### PR TITLE
chore: Update dependencies and add discovery with testing

### PR DESCRIPTION
**Pull request by cedricdcc**
this branch will fix the following issues:

- #58

### CODE REVIEW

File: sema/discovery/url_to_graph.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            finally:
+                return graph
+
+    if not triples_found:
+        # perform a check in the html to
+        # see if there is any link to fair signposting
+        # perform request to uri with accept header text/html
+        headers = {"Accept": "text/html"}
+        r = session.get(subject_url, headers=headers)
+        if r.status_code == 200 and "text/html" in r.headers["Content-Type"]:
```
Comment (marc-portier) [inline sema/discovery/url_to_graph.py:99]: make an issue to enhance this kind of tests -- I rather see us using a lib-function like [`cgi.parse_header()`](https://docs.python.org/3/library/cgi.html) for proper http header parser - allowing actual match to the mimetype, and not an unintendent (but unlikely) stray match

File: tests/discovery/test_url_to_graph.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    assert ctype_to_rdf_format("application/json") is None
+    assert ctype_to_rdf_format("application/octet-stream") is None
+    assert ctype_to_rdf_format("text/html") is None
+
+
+# ttl file
+# jsonld file
+# html file with ttl file in head as full script
+# html file with jsonld file in head as script link
+test_cases = [
```
Comment (marc-portier) [inline tests/discovery/test_url_to_graph.py:26]: these tests and assertions heavily depend on external services
I would rather see us push up a local http server during tests to better control their outcome 
no need to that now, but best to track an issue for this

**Review by marc-portier — APPROVED** [review]
ready to go - please add some issues for later on the remarks made - thx

File: tests/discovery/test_url_to_graph.py (hunk 2 above)
Comment (cedricdcc) [reply tests/discovery/test_url_to_graph.py:26]: Issue taken up in #66

File: sema/discovery/url_to_graph.py (hunk 1 above)
Comment (cedricdcc) [reply sema/discovery/url_to_graph.py:99]: Issue taken up in #65

**Pull request merged**